[error_category Category
    FAIL
    CRASH
]

[extension_method toString Category @[utf8]()
    [switch [this]
        [case FAIL
            [result] := "failure!"
        ][case CRASH
            [result] := "crash!"
        ]
    ]
]

[function doSomethingBad ![void]()
    [throw Category..FAIL]
]

[entry_point
    doSomethingBad().[catch 
            Category
            std..print(line:"failed because of error: {[error]->toString()}.")
        ]
]